azure-pipelines.yml

```diff
@@ -6,108 +6,106 @@ trigger:
+- ${{ else }}:
+  - name: PostBuildSign
+    value: true
 
-  # Produce test-signed build for PR and Public builds
-  - ${{ if or(eq(variables['System.TeamProject'], 'public'), in(variables['Build.Reason'], 'PullRequest')) }}:
-    - name: SignType
-      value: test
+# Set up non-PR build from internal project
+- name: SignType
+  value: $[ coalesce(variables.OfficialSignType, 'real') ]
 
-  # Set up non-PR build from internal project
-  - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
-    - name: SignType
-      value: $[ coalesce(variables.OfficialSignType, 'real') ]
+resources:
+  repositories:
+  - repository: 1ESPipelineTemplates
+    type: git
+    name: 1ESPipelineTemplates/1ESPipelineTemplates
+    ref: refs/tags/release
+extends:
+  template: v1/1ES.Official.PipelineTemplate.yml@1ESPipelineTemplates
+  parameters:
+    featureFlags:
+      autoBaseline: true
+    pool:
+      name: NetCore1ESPool-Internal
+      image: 1es-windows-2022-pt
+      os: windows
+    customBuildTags:
+    - ES365AIMigrationTooling
+    stages:
+    - stage: Build
+      jobs:
 
-stages:
-- stage: Build
-  jobs:
+      # Windows x64
+      - template: /eng/pipelines/jobs/windows-build.yml@self
+        parameters:
+          name: win_x64
+          targetArchitecture: x64
 
-  # Windows x64
-  - template: /eng/pipelines/jobs/windows-build.yml
-    parameters:
-      name: win_x64
-      targetArchitecture: x64
+      # Windows x86
+      - template: /eng/pipelines/jobs/windows-build.yml@self
+        parameters:
+          name: win_x86
+          targetArchitecture: x86
 
-  # Windows x86
-  - template: /eng/pipelines/jobs/windows-build.yml
-    parameters:
-      name: win_x86
-      targetArchitecture: x86
+      # Windows arm64
+      - template: /eng/pipelines/jobs/windows-build.yml@self
+        parameters:
+          name: win_arm64
+          targetArchitecture: arm64
 
-  # Windows arm64
-  - template: /eng/pipelines/jobs/windows-build.yml
-    parameters:
-      name: win_arm64
-      targetArchitecture: arm64
+    - stage: PrepareForPublish
+      displayName: Prepare for Publish
+      dependsOn: Build
+      jobs:
+      # Prep artifacts: sign them and upload pipeline artifacts expected by stages-based publishing.
+      - template: /eng/pipelines/jobs/prepare-signed-artifacts.yml@self
+        parameters:
+          PublishRidAgnosticPackagesFromJobName: win_x64
+      # Publish to Build Asset Registry in order to generate the ReleaseConfigs artifact.
+      - template: /eng/common/templates-official/job/publish-build-assets.yml@self
+        parameters:
+          publishUsingPipelines: true
+          dependsOn: PrepareSignedArtifacts
+          pool:
+            name: NetCore1ESPool-Internal
+            demands: ImageOverride -equals windows.vs2022preview.amd64
 
-- ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
```

Comment:
Is there guidance on how to do this correctly and does the same need to be done for winforms/wpf?

Reply:
Same needs to be done for wpf/winforms (since they take internal PRs). For each of these types of conditionals, you can say the following
- The pr pipeline can assume PR == true. Often this has little do with actual pull requests, and more to do with what testing is run, or whether real signing should be attempted, for instance.
- The PR pipeline should not assume public vs. internal.

Does that answer your question?

Reply:
Apologies, I wasn't specific, I mean guidance in terms of creating the PR pipeline on the internal project. Specifically any naming or space to update that tracks the new pipelines needed to be created (we can also chat offline that may be better 😄 )